src/rpc/mining.cpp

```diff
@@ -1172,21 +1196,32 @@ UniValue submitblock(const UniValue& params, bool fHelp)
     return BIP22ValidationResult(state);
 }
 
-UniValue validateBlock(const UniValue& params, bool fHelp)
+UniValue validateBlocks(const UniValue& params, bool fHelp)
 {
-    const CBlock b2 = blockFromUnivalue(params);
+    std::vector<CBlock> blocks = blocksFromUnivalue(params);
     CValidationState state;
     CBlockIndex* const pindexPrev = chainActive.Tip();
     // TestBlockValidity only supports blocks built on the current Tip
-    uint256 thisPrevBlock = b2.hashPrevBlock;
-    CBlockHeader thisHeader = b2.GetBlockHeader();
+    uint256 thisPrevBlock = blocks[0].hashPrevBlock;
+    CBlockHeader thisHeader = blocks[0].GetBlockHeader();
     uint256 thisHash = thisHeader.GetHash();
     uint256 actualPrevBlock = pindexPrev->GetBlockHash();
     if (thisHash == actualPrevBlock)
         return "i-am-block-producer";
-    if (thisPrevBlock != actualPrevBlock)
+    if (thisPrevBlock != actualPrevBlock) {
```

Comment:
we should get rid of this if and begin the iteration at the first block and let the logic of this if be encapsulated by the if statement within the loop `if (desc->hashPrevBlock == actualPrevBlock) {`. In other words we shouldn't have a special case for: the most recent block is the only block to validate. That case should be hit just like all our other cases.

Reply:
You're right! fixed.